### ISSUE TITLE
A sudden pause at STEP 7

### ISSUE THREAD

**Issue by judyyyyhsgfh** [broadinstitute/infercnv#484]
Hello, thank you for developing and maintaining this wonderful tool.

I tried to get the output file "17_HMM_predHMMi6.hmm_mode-samples.cell_groupings" containing CNV classifications, but the process paused at STEP 7 without warnings or errors.

Here are my codes :
‘
infercnv_obj = CreateInfercnvObject(raw_counts_matrix=exprMatrix,
                                    annotations_file=" groupFiles.txt",
                                    delim="\t",
                                    gene_order_file= "geneLocate.txt",
                                    ref_group_names="0")

infercnv_obj = infercnv::run(infercnv_obj,
                             cutoff=1, 
                             out_dir="output1202",
                             cluster_by_groups=TRUE, 
                             tumor_subcluster_partition_method='random_trees',
                             tumor_subcluster_pval=0.05,
                             analysis_mode="subclusters",
                             denoise=TRUE,
                             HMM=TRUE,
                             num_threads=4)
’

It paused here as the picture shows, and the occupations of CPU and storage went down. 
<img width="748" alt="image" src="https://user-images.githubusercontent.com/119640909/205256490-d6538686-bb8d-47bb-9d62-75a234e1249a.png">

Version informations: 
R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)
infercnv: 1.12.0

Thank you so much for your help!
Judy

**Comment by GeorgescuC** [comment]
Hi @judyyyyhsgfh ,

The random trees subclustering method, which is the step you are at, can take a while to run, especially if your dataset has over a couple thousand cells. If it takes too long to run, you can try the default subclustering method, Leiden, which is a lot faster and has more fine tuning options available.

Regards,
Christophe.

**Comment by judyyyyhsgfh** [comment]
Thank you so much for your reply!

I tried Leiden at first. It seems that subcluster

**Comment by judyyyyhsgfh** [comment]
Hi @GeorgescuC , thank you so much for your reply!

I tried the default analysis mode at first (without codes 'analysis_mode="subclusters" '). It seems like that there's no branching information about subclusters in the output file "17_HMM_predHMMi6.hmm_mode-samples.cell_groupings", and that's what I tried to get. I ran codes above on another computer with larger RAM (16GB to 128GB), and got all output files in about 10 hours.
